client/src/components/item-details/ItemDetails.tsx

```diff
@@ -1,13 +1,20 @@
+// import mock from '../../mock-data/mock.json'
+
 import styles from './ItemDetails.module.css'
 import { useCartSlice } from '../../zustand/ShoppingCartSlice'
-import mock from '../../mock-data/mock.json'
 import { useParams } from 'react-router-dom'
-import { Product, ProductSize } from '../../models/models'
+import { ProductSize } from '../../models/models'
 import { useState } from 'react'
 import { addToShoppingCart } from '../../services/shopping-cart-service'
+import { useProductsSlice } from '../../zustand/ProductSlice'
+import { userStore } from '../../zustand/UserStore'
+import { ShoppingCartProductType } from '../../../../global-types/shopping-cart-product'
 
 
 export default function ItemDetails() {
+  const id = userStore((state) => state.id)
```

Comment:
consider using object destructuring to make the code less verbose:
const { id } = userStore() gives the same result.